# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: e495155bbdd1
# Full Hash: e495155bbdd166ce20c531f9185a42af1833734d
# Author: Brendan Dahl <bdahl@mozilla.com>
# Date: 2019-05-29 21:50:20
# Description:
#   Bug 1554072 - Only crash in debug builds when using createElement with XUL. r=bzbarsky
#   
#   Temporary solution until we get sanitized chrome JS stacks for the crash
#   reporter.
#   
# ==============================================================================

diff -r 652a6bf1c627 -r e495155bbdd1 dom/base/Document.cpp
--- a/dom/base/Document.cpp	Wed May 29 15:21:52 2019 +0000
+++ b/dom/base/Document.cpp	Wed May 29 01:31:11 2019 +0000
@@ -5545,8 +5545,7 @@
 #if DEBUG
     xpc_DumpJSStack(true, true, false);
 #endif
-    MOZ_DIAGNOSTIC_ASSERT(false,
-                          "CreateElement() not allowed in XUL document.");
+    MOZ_ASSERT_UNREACHABLE("CreateElement() not allowed in XUL document.");
     rv.Throw(NS_ERROR_FAILURE);
     return nullptr;
   }
